src/main/groovy/org/savantbuild/plugin/java/testng/JavaTestNGPlugin.groovy

```diff
@@ -139,6 +153,23 @@ class JavaTestNGPlugin extends BaseGroovyPlugin {
       } else {
         output.infoln("No test results found from a prior test run. File not found [" + testResults.toString() + "].")
       }
+    } else if (runtimeConfiguration.switches.booleanSwitches.contains("onlyChanges")) {
+      output.infoln("Only run tests for changed files (⚠️ misses changes in dependencies).")
+      String numberCommitsFlag = ""
+      if (runtimeConfiguration.switches.valueSwitches.containsKey("n")) {
+        numberCommitsFlag = "-n " + runtimeConfiguration.switches.values("n").first()
```

Comment:
should we do number of commits or just take a baseline ref? Meaning it would diff against `main` if you supplied `main`.

Reply:
I ended up reworking the logic to first try `gh pr diff` and then fall back to `git diff`.  Now, instead of supplying `--n`, you can supply `--commitRange` and tell git exactly what to compare.